sql/atxs/atxs.go

```diff
@@ -873,46 +874,26 @@ func IterateAtxIdsWithMalfeasance(
 	return err
 }
 
-type PrevATXCollision struct {
-	NodeID1 types.NodeID
-	ATX1    types.ATXID
-
-	NodeID2 types.NodeID
-	ATX2    types.ATXID
-}
-
-func PrevATXCollisions(db sql.Executor) ([]PrevATXCollision, error) {
-	var result []PrevATXCollision
-
+func PrevATXCollision(db sql.Executor, prev types.ATXID, id types.NodeID) (types.ATXID, types.ATXID, error) {
+	var atxs []types.ATXID
+	enc := func(stmt *sql.Statement) {
+		stmt.BindBytes(1, prev[:])
+		stmt.BindBytes(2, id[:])
+	}
 	dec := func(stmt *sql.Statement) bool {
-		var nodeID1, nodeID2 types.NodeID
-		stmt.ColumnBytes(0, nodeID1[:])
-		stmt.ColumnBytes(1, nodeID2[:])
-
-		var id1, id2 types.ATXID
-		stmt.ColumnBytes(2, id1[:])
-		stmt.ColumnBytes(3, id2[:])
-
-		result = append(result, PrevATXCollision{
-			NodeID1: nodeID1,
-			ATX1:    id1,
-
-			NodeID2: nodeID2,
-			ATX2:    id2,
-		})
-		return true
+		var id types.ATXID
+		stmt.ColumnBytes(0, id[:])
+		atxs = append(atxs, id)
+		return len(atxs) < 2
 	}
-	// we are joining the table with itself to find ATXs with the same prevATX
-	// the WHERE clause ensures that we only get the pairs once
-	if _, err := db.Exec(`
-		SELECT p1.pubkey, p2.pubkey, p1.atxid, p2.atxid
-		FROM posts p1
-		INNER JOIN posts p2 ON p1.prev_atxid = p2.prev_atxid
-		WHERE p1.atxid < p2.atxid;`, nil, dec); err != nil {
-		return nil, fmt.Errorf("error getting ATXs with same prevATX: %w", err)
+	_, err := db.Exec("SELECT atxid FROM posts WHERE prev_atxid = ?1 AND pubkey = ?2;", enc, dec)
```

Comment:
Should we add sorting by timestamp (oldest first) so that we always get the same 2 ATXs in the same query?

Reply:
Sure we can add ordering but it will make the query slower.